packages/cojson/src/coValues/coMap.ts

```diff
@@ -310,10 +318,13 @@ export class RawCoMapView<
       }
     | undefined {
     const ops = this.timeFilteredOps(key);
-    if (!ops || ops.length === 0) {
+    const lastEntry = ops?.[ops.length - 1];
+
+    if (!lastEntry) {
       return undefined;
     }
-    return this.nthEditAt(key, ops.length - 1);
+
+    return operationToEditEntry(lastEntry);
```

Comment:
Avoided to call `nthEditAt` since we already have the entry that we want to return.

This can be further optimized by removing `timeFilteredOps` but didn't want to complicate this function too much.